### PR TITLE
Document `Partition` exceptions

### PR DESCRIPTION
**Pull request by viceroypenguin**

### CODE REVIEW

**Comment by viceroypenguin** [comment]
NB: requires #881 to be merged first

File: MoreLinq/Extensions.g.cs [hunk 1]
```diff
@@ -4154,6 +4154,7 @@ public static partial class PartitionExtension
         /// A tuple of elements satisfying the predicate and those that do not,
         /// respectively.
         /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null</exception>
```
Comment (atifaziz) [inline MoreLinq/Extensions.g.cs:4157]: Use `<see langword="null"/>` instead of `null` consistently for new updates to docs?

Comment (viceroypenguin) [reply MoreLinq/Extensions.g.cs:4157]: Good catch!

Comment (atifaziz) [inline MoreLinq/Extensions.g.cs:4157]:
```suggestion
        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
```

**Review by atifaziz — APPROVED** [review]

File: MoreLinq/Partition.cs [hunk 2]
```diff
@@ -33,6 +33,7 @@ static partial class MoreEnumerable
         /// A tuple of elements satisfying the predicate and those that do not,
         /// respectively.
         /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null</exception>
```
Comment (atifaziz) [inline MoreLinq/Partition.cs:36]:
```suggestion
        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
```

**Pull request merged**